### PR TITLE
arm: allow nesting in DisableInterrupts and EnableInterrupts 

### PR DESCRIPTION
**Pull request by aykevl**
This finally fixes a TODO left in the code.

### CODE REVIEW

**Comment by aykevl** [comment]
I tested locally with the following code:

```go
	mask := arm.DisableInterrupts()
	if true {
		mask := arm.DisableInterrupts()
		arm.EnableInterrupts(mask)
	}
	ws.WriteColors(strip)
	arm.EnableInterrupts(mask)
```

Before this PR, it flickers if the condition is true. Afterwards, it is flicker-free. Tested on a Circuit Playground Express connected to a computer (and thus generating random interrupts).

File: src/device/avr/avr.go [hunk 1]
```diff
@@ -15,4 +15,4 @@ func Asm(asm string)
 //             "value":  1
 //             "result": &dest,
 //         })
-func AsmFull(asm string, regs map[string]interface{})
+func AsmFull(asm string, regs map[string]interface{}) uintptr
```
Comment (jaddr2line) [inline src/device/avr/avr.go:18]: The return value should be documented in the comment.

File: src/device/riscv/riscv.go [hunk 2]
```diff
@@ -5,6 +5,14 @@ package riscv
...
+// effects, as it would otherwise be optimized away. The inline assembly string
+// recognizes template values in the form {name}, like so:
+//
+//     arm.AsmFull(
+//         "st {value}, {result}",
+//         map[string]interface{}{
+//             "value":  1
+//             "result": &dest,
+//         })
+func AsmFull(asm string, regs map[string]interface{}) uintptr
```
Comment (jaddr2line) [inline src/device/riscv/riscv.go:18]: The return value should be documented in the comment.

**Comment by aykevl** [comment]
@jaddr2line thanks, fixed.

**Comment by deadprogram** [comment]
OK merging. Thanks!

**Pull request merged**